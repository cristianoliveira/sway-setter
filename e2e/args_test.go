package e2e

import (
	"testing"

	"github.com/cristianoliveira/sway-setter/internal/testutils"
	"github.com/gkampitakis/go-snaps/snaps"
)

func TestFlags(t *testing.T) {
	cases := []struct {
		name       string
		args       []string
		expectFail bool
	}{
		{
			name:       "no flag",
			args:       []string{},
			expectFail: true,
		},
		{
			name: "help flag",
			args: []string{
				"-h",
			},
			expectFail: false,
		},
		{
			name: "help flag",
			args: []string{
				"--help",
			},
			expectFail: false,
		},
	}

	for _, tc := range cases {
		t.Run(tc.name, func(tt *testing.T) {
			cmd := testutils.NewCommandTesting("", tc.args...)

			err := cmd.Run()
			if err != nil && !tc.expectFail {
				tt.Fatalf(err.Error())
			}

			command, err := cmd.String()
			if err != nil {
				tt.Fatalf(err.Error())
			}

			snaps.MatchSnapshot(tt, command)
		})
	}
}
