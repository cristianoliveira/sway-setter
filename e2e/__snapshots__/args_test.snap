
[TestFlags/no_flag:_[] - 1]
../bin/sway-setter
Usage:
  sway-setter [OPTIONS]

Application Options:
  -t, --type=  Type 'set_{type}' object to be set analog of 'swaymsg -t
               get_{type}'
  -p, --print  Prints commands that would be executed. Can be used as input to
               swaymsg

Help Options:
  -h, --help   Show this help message


---

[TestFlags/help_flag:_[-h] - 1]
../bin/sway-setter -h
Usage:
  sway-setter [OPTIONS]

Application Options:
  -t, --type=  Type 'set_{type}' object to be set analog of 'swaymsg -t
               get_{type}'
  -p, --print  Prints commands that would be executed. Can be used as input to
               swaymsg

Help Options:
  -h, --help   Show this help message


---

[TestFlags/help_flag:_[--help] - 1]
../bin/sway-setter --help
Usage:
  sway-setter [OPTIONS]

Application Options:
  -t, --type=  Type 'set_{type}' object to be set analog of 'swaymsg -t
               get_{type}'
  -p, --print  Prints commands that would be executed. Can be used as input to
               swaymsg

Help Options:
  -h, --help   Show this help message


---
